src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -0,0 +1,444 @@
+                            .collect(toMap(Channel::getKey, Channel::getId))
+                )
+                .thenCompose(v -> createMissingSupplyChannels(drafts));
+    }
+
+    /**
+     * When {@code ensureChannel} from {@link InventorySyncOptions} is set to {@code true} then attempts to create
+     * missing supply channels. Missing supply channel is a supply channel of key that can not be found in CTP project,
+     * but occurs in {@code drafts} list. Method returns {@link CompletionStage} of {@link Void} that indicates all
+     * possible creation attempts progress.
+     *
+     * @param drafts {@link List} containing {@link InventoryEntryDraft} objects where missing supply channels can occur
+     * @return {@link CompletionStage} of {@link Void} that indicates all possible creation attempts progress
+     */
+    @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
+    private CompletionStage<Void> createMissingSupplyChannels(@Nonnull final List<InventoryEntryDraft> drafts) {
+        if (syncOptions.shouldEnsureChannels()) {
+            final List<String> missingChannelsKeys = drafts.stream()
+                .map(SkuKeyTuple::of)
+                .map(SkuKeyTuple::getKey)
+                .distinct()
+                .filter(Objects::nonNull)
+                .filter(key -> !supplyChannelKeyToId.containsKey(key))
+                .collect(toList());
+            final List<CompletableFuture<Void>> creationStages = missingChannelsKeys.stream()
+                .map(this::createMissingSupplyChannel)
+                .map(CompletionStage::toCompletableFuture)
+                .collect(toList());
+            return CompletableFuture.allOf(creationStages.toArray(new CompletableFuture[creationStages.size()]));
+        } else {
+            return CompletableFuture.completedFuture(null);
+        }
+    }
+
+    /**
+     * Fetches existing {@link InventoryEntry} objects from CTP project that correspond to passed {@code batchOfDrafts}.
+     * Having existing inventory entries fetched, {@code batchOfDrafts} is compared and synced with fetched objects by
+     * {@link InventorySync#compareAndSync(Map, List)} function. When fetching existing inventory entries results in
+     * exception then error callback is executed and {@code batchOfDrafts} isn't processed.
+
+     * @param batchOfDrafts batch of drafts that need to be synced
+     * @return {@link CompletionStage} of {@link Void} that indicates method progress.
+     */
+    private CompletionStage<Void> processBatch(final List<InventoryEntryDraft> batchOfDrafts) {
+        return fetchExistingInventories(batchOfDrafts)
+                .thenCompose(existingInventories -> compareAndSync(existingInventories, batchOfDrafts))
+                .exceptionally(ex -> {
+                    handleFailure(format("Failed to find existing inventory entries of SKUs %s",
```

Comment:
Exception handling will be adjusted further with https://github.com/commercetools/commercetools-sync-java/issues/31 and https://github.com/commercetools/commercetools-sync-java/issues/30 issues